### PR TITLE
API: Add user-friendly methods for INCR/DECR and ZRANK/ZREVRANK

### PR DESCRIPTION
**Pull request by Mimi8298**
Add new methods to the API to simplify its use:
- SortedSetRank allows ZRANK/ZREVRANK to be called via key and member 
- Increment allows INCR/DECR/INCRBY/DECRBY to be called via key and increment number. OperationError is returned but I don't know if this is the best way to return it? I've seen that other methods simply return a string as an error

### CODE REVIEW

**Comment by Mimi8298** [comment]
@microsoft-github-policy-service agree

**Comment by Mimi8298** [comment]
Is it possible to re-run the action?
Something went wrong at the results upload (github issue)

**Review by TalZaccai — CHANGES_REQUESTED** [review]
Still waiting for changes, any updates? :)

File: libs/server/Storage/Session/ObjectStore/SortedSetOps.cs [hunk 1]
```diff
@@ -676,6 +676,45 @@ public unsafe GarnetStatus SortedSetScan<TObjectContext>(ArgSlice key, long curs
 
         }
 
+        /// <summary>
+        /// Returns the rank of member in the sorted set, the scores in the sorted set are ordered from high to low
+        /// </summary>
+        public unsafe GarnetStatus SortedSetRank<TObjectContext>(ArgSlice key, ArgSlice member, bool reverse, out long? rank, ref TObjectContext objectStoreContext)
```
[the remark this answers is not in the document]
Comment (Mimi8298) [reply libs/server/Storage/Session/ObjectStore/SortedSetOps.cs:681]: Currently, no API tests exist here (. I can eventually add tests in a new file or add them to RespTransactionProcTests?

File: libs/server/Storage/Session/MainStore/MainStoreOps.cs [hunk 2]
```diff
@@ -834,6 +834,51 @@ public GarnetStatus Increment<TContext>(ArgSlice key, ArgSlice input, ref ArgSli
             return GarnetStatus.OK;
         }
 
+        public unsafe GarnetStatus Increment<TContext>(ArgSlice key, long increment, out OperationError operationError, out long output, ref TContext context)
```
[the remark this answers is not in the document]
Comment (Mimi8298) [reply libs/server/Storage/Session/MainStore/MainStoreOps.cs:837]: Same, no API tests exist here :)

[the remark this answers is not in the document]
Comment (Mimi8298) [reply libs/server/Storage/Session/MainStore/MainStoreOps.cs]: Done, I have removed the OperationError and added the assert.

File: libs/server/API/IGarnetApi.cs [hunk 3]
```diff
@@ -185,6 +185,16 @@ public interface IGarnetApi : IGarnetReadApi, IGarnetAdvancedApi
         /// <param name="output"></param>
         /// <returns></returns>
         GarnetStatus Increment(ArgSlice key, ArgSlice input, ref ArgSlice output);
+
+        /// <summary>
+        /// Increment (INCR, INCRBY, DECR, DECRBY)
```
[the remark this answers is not in the document]
Comment (Mimi8298) [reply libs/server/API/IGarnetApi.cs]: Done, I have added a default value incrementCount = 1 and created a new method `GarnetStatus Decrement(ArgSlice key, out long output, long decrementCount = 1)`.

File: libs/server/Storage/Session/ObjectStore/SortedSetOps.cs [hunk 4]
```diff
@@ -676,6 +676,45 @@ public unsafe GarnetStatus SortedSetScan<TObjectContext>(ArgSlice key, long curs
 
         }
 
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (Mimi8298) [reply libs/server/Storage/Session/ObjectStore/SortedSetOps.cs:678]: Done

File: libs/server/API/IGarnetApi.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (TalZaccai) [reply libs/server/API/IGarnetApi.cs]: Great!

File: libs/server/Storage/Session/ObjectStore/SortedSetOps.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TalZaccai) [reply libs/server/Storage/Session/ObjectStore/SortedSetOps.cs:686]: Ohhh sorry I just realized what you were asking, you mean that GarnetClient doesn't have a method that calls these specific methods... Yeah then let's leave it at that for now.

**Review by TalZaccai — APPROVED** [review]

**Pull request merged**